cmd/zfs/zfs_projectutil.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+/*
+ * Copyright (c) 2017, Intel Corporation. All rights reserved.
+ */
+
+#ifndef	_ZFS_PROJECTUTIL_H
+#define	_ZFS_PROJECTUTIL_H
+
+typedef enum {
+	ZFS_PROJECT_INVALID	= 0,
```

Comment:
Should be consided as valid.

Reply:
"ZFS_PROJECT_INVALID" is the default "zfs project" operation type. We need to specify either "-s", or "-C" or "-c" or "-p" for "zfs project" command. If non of them is specified, the operation type will be ZFS_PROJECT_INVALID. Then the operation will fail.